Source/ReactiveCore/MoyaError.swift

```diff
@@ -9,3 +9,59 @@ public enum MoyaErrorCode: Int {
     case StatusCode
     case Data
 }
+
+public enum MoyaError: ErrorType {
+    case ImageMapping(MoyaResponse)
+    case JSONMapping(MoyaResponse)
+    case StringMapping(MoyaResponse)
+    case StatusCode(Int,MoyaResponse)
+    case Data(MoyaResponse)
+    case Underlying(ErrorType)
+    
+    public func sameErrorType(otherError: MoyaError) -> Bool {
```

Comment:
Where there were a better way to structure this function, but I don't think there is. Is it useful to developers using Moya, or just us when testing? Either way, I think it might look cleaner inside an extension.